lib/surro-gate/proxy.rb

```diff
@@ -111,16 +111,21 @@ def thread_stop
     end
 
     def reactor
-      # Atomically get an array of readable/writable monitors
-      monitors = @mutex.synchronize { @selector.select(0.1) || [] }
+      # Atomically get an array of readable monitors while also polling for writables
+      monitors = @mutex.synchronize do
+        @writer.select(0.1)
+        @reader.select(0.1) || []
+      end
       # Call each transmission proc and collect the results
-      callers = monitors.map { |m| m.value.call }
-      # Sleep for a short time if there was no transmission
-      sleep(0.1) if callers.none? && monitors.any?
+      monitors.map { |m| m.value.call }
+      # Sleep for a short time if there are no sockets
+      sleep(0.1) if @reader.empty? && @writer.empty?
```

Comment:
This `0.1s` happens only if there are no opened sockets for proxying. The delay would be sensible only when the first connection is opened. However, the internal thread is not running if there are no registered connections. As the thread is killed if the last connection gets closed, there is a high chance that this sleep gets never executed. I just simply don't see enough into Ruby's threads to be 100% sure that we can delete this line :confused: